src/librustc_borrowck/borrowck/unused.rs

```diff
@@ -77,7 +77,7 @@ impl<'a, 'tcx> UnusedMutCx<'a, 'tcx> {
                 continue
             }
 
-            let mut_span = tcx.sess.codemap().span_until_char(ids[0].2, ' ');
+            let mut_span = tcx.sess.codemap().span_through_char(ids[0].2, ' ');
```

Comment:
What if there is more than one space after the `mut`? Also, and this is a bug with the existing code as well, what if the white-space after the `mut` doesn't contain any spaces at all?

Reply:
The following highlights until the end of `foo`.

```
fn main() {
    let mut
foo = 1;
    println!("{}", foo);
}
```

I don't think multiple spaces are an issue, because there'll be the same number of multiple spaces as there were before.

Reply:
This is a blocking issue. I'll take a look at it.

Reply:
The code now accounts for _any_ whitespace characters, in any amount, so that the resulting code will always be one space. I could see this breaking formatting sometimes, but should never produce incorrect code.